tests/integration-tests/tests/catalog_tests.rs

```diff
@@ -142,7 +142,7 @@ async fn test_catalog_lookup() {
     let question_req = Request::new(question_req, ([127, 0, 0, 1], 5553).into(), Protocol::Udp);
 
     let response_handler = TestResponseHandler::new();
-    catalog
+    let _ = catalog
```

Comment:
I think you should just `unwrap()` here since it's a test and you (probably) expect the call to be successful